src/data/nat/factorization.lean

```diff
@@ -43,12 +43,16 @@ namespace nat
  mapping each prime factor of `n` to its multiplicity in `n`. -/
 noncomputable def factorization (n : ℕ) : ℕ →₀ ℕ := (n.factors : multiset ℕ).to_finsupp
 
+/-! ### Basic facts about factorization -/
+
 @[simp] lemma factorization_prod_pow_eq_self {n : ℕ} (hn : n ≠ 0) : n.factorization.prod pow = n :=
 begin
   simp only [←prod_to_multiset, factorization, multiset.coe_prod, multiset.to_finsupp_to_multiset],
   exact prod_factors hn,
 end
 
+/-- Since `factorization` is a finsupp it's noncomputable.  This theorem allows us to
+obtain values of `factorization n p` when required. -/
```

Comment:
```suggestion
/-- 
We can write both `n.factorization p` and `n.factors.count p` to represent the power
of `p` in the factorization of `n`: we declare the former to be simp-normal form.
However, since `factorization` is a finsupp it's noncomputable.  This theorem can also
be used in reverse to compute values of `factorization n p` when required. -/
```

Reply:
Or something along these lines!